File: TCS_NQT/problem_2/average.cpp
#include <iostream>
#include <numeric> 
using namespace std;
template <size_t N>

double average(const int (&arr)[N]) {
    return accumulate(*&arr, *&arr + N, 0.0) / N; 
}


int main() {
    int arr1[] = {1, 2, 3, 4, 5};
    int arr2[] = {1, 2, 1, 1, 5, 1};

    cout << "Average of arr1: " << average(arr1) << endl;
    cout << "Average of arr2: " << average(arr2) << endl;

    return 0;
}

File: TCS_NQT/problem_2/average.java
import java.util.*;

public class average {
    /*
     * using array stream for average calculation
     */
    public static double average(int arr[]) {
        return Arrays.stream(arr).average().orElse(0);
    }

    public static void main(String[] args) {
        // Test cases
        int arr1[] = {1, 2, 3, 4, 5};
        int arr2[] = {1, 2, 1, 1, 5, 1};

        System.out.println(average(arr1)); 
        System.out.println(average(arr2)); 
    }
}
